cmd/modelfile/generate.go

```diff
@@ -75,7 +112,44 @@ func init() {
 }
 
 // runGenerate runs the generate modelfile.
-func runGenerate(_ context.Context) error {
+func runGenerate(ctx context.Context) error {
+	// If model URL is provided, download the model first
+	if generateConfig.ModelURL != "" {
+		fmt.Printf("Model URL provided: %s\n", generateConfig.ModelURL)
+
+		// Get the appropriate provider for this URL
+		registry := modelprovider.NewRegistry()
+		provider, err := registry.GetProvider(generateConfig.ModelURL)
+		if err != nil {
+			return fmt.Errorf("unsupported model URL: %w", err)
+		}
+
+		fmt.Printf("Using provider: %s\n", provider.Name())
+
+		// Check if user is authenticated with the provider
+		if err := provider.CheckAuth(); err != nil {
+			return fmt.Errorf("%s authentication check failed: %w", provider.Name(), err)
+		}
+
+		// Create a temporary directory for downloading the model
+		// Clean up the temporary directory after the function returns
+		tmpDir, err := os.MkdirTemp("", "modctl-model-downloads-*")
```

Comment:
Do we need to consider expose this temp dir to user? because in some limited environment, user may only have write access to specific dirs.